checkstyle-tester/diff.groovy

```diff
@@ -218,11 +220,11 @@ def launchCheckstyleReport(cfg) {
 
     // If "no exception" testing, these may not be defined in repos other than checkstyle
     if (isRegressionTesting) {
-        println "Installing Checkstyle artifact ($cfg.branch) into local Maven repository ..."
+        println "Building Checkstyle artifact ($cfg.branch) ..."
         executeCmd("git checkout $cfg.branch", cfg.localGitRepo)
         executeCmd("git log -1 --pretty=MSG:%s%nSHA-1:%H", cfg.localGitRepo)
-        executeCmd("mvn -e --no-transfer-progress --batch-mode -Pno-validations clean install",
-            cfg.localGitRepo)
+        executeCmd("mvn -e --no-transfer-progress --batch-mode -Pno-validations clean package -Passembly " +
+            "-Dassembly.skipAssembly=true -Dmaven.test.skip=true", cfg.localGitRepo)
```

Comment:
`-Pno-validations` atleast skips tidy.
`-Dassembly.skipAssembly=true` skips `maven-assembly-plugin` which builds tar/zip bin/src which is not needed for regression.
`-Dmaven.test.skip=true` skips test resources, compiling, and jar which is also not needed for regression (see https://github.com/checkstyle/contribution/pull/747#discussion_r1059673537 )

Reply:
This update shows gaps in our profile
https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L2174
It should skip tidy.

Reply:
We should add all other skips that you need to this profile, as the only goal of this profile is to get Uber jar

Reply:
Why do we combine no-validation with assembly? We can keep them separate so we don't have to worry about constantly synching, and we can still combine them both for things like this.

Reply:
We should not combine, I never did this.

Reply:
Do you think we need to make them to not overlap, and use them as combination?

Reply:
I think it will be easier to remember and maintain if each profile just have 1 main job. Sevntu does this at https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/.ci/validation.sh#L50

Reply:
Ok let's combine no-validation and assembly.

But assembly should skip tests jaring and assembly plugin. So all 2 parameters should be part of assembly profile.

Does it make sense ?

Reply:
@rnveach , is there concerns ?

Reply:
I combined the 2 profiles into 1 command. `-Pno-validations,assembly`.

I made changes to the profiles in main repo. Let me know if I missed something, otherwise I think this is done.